### ISSUE TITLE
Fix some bugs

### ISSUE THREAD

**Issue by violet-zct** [violet-zct/pytorch_NMT#13]
Hi guys,

I caught two bugs in our code, you can check FIXME in the code to find where I fixed.
These are subtle bugs which mainly involve wrong indexing:
(1) When using delta rewards, their indices don't with the actual sample index, since the first added token in the sample is not what the sampler generates.
(2) Similarly, the second bug is we are using the wrong mask before.

These bugs won't result in fatal errors however we need to rerun the experiments to get the right results. 

Besides, Yifan, can you run a new baseline MLE model with encoder and decoder the same hidden dimensions, I have changed that in the __init__() function, so you can directly run.

I found MIXER model is a little bit trickier than I thought mainly because we should be very careful dealing with the index and concatenating the cross-entropy part and RL part. The code I just committed is not finished yet, I am debugging the MIXER part and let you know when I finish.

Chunting
